org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/MavenImpl.java

```diff
@@ -1280,4 +1281,19 @@ public MavenExecutionContext getExecutionContext() {
     return MavenExecutionContext.getThreadContext();
   }
 
+  /* (non-Javadoc)
+   * @see org.apache.maven.Maven#execute(org.apache.maven.execution.MavenExecutionRequest)
+   */
+  @Override
+  public MavenExecutionResult execute(MavenExecutionRequest request) {
+    MavenExecutionContext context = new MavenExecutionContext(this, request);
+    try {
+      return context.execute((innerContext, monitor) -> {
+        return innerContext.getSession().getResult();
+      }, new NullProgressMonitor());
+    } catch(CoreException ex) {
+      throw new RuntimeException(ex);
```

Comment:
The following is from the 'old' implementation and I think it is more suitable than throwing a bare RuntimeException.
```suggestion
      return new DefaultMavenExecutionResult().addException(ex);
```